lib/scanners/base.js

```diff
@@ -34,17 +34,41 @@ var BaseScanner = function (beaconMapper, beaconRegionMapper, beaconRegionMonito
     this.init = function (networkIdentifier) {
         if (!_.isUndefined(networkIdentifier)) {
             self.networkId = networkIdentifier;
+            var hasPermissions = function () {
+                beaconHandler.init();
+                beaconLog.init();
+                self.setBeaconRegions([]);
+                if (OS_ANDROID || (OS_IOS && (_.isUndefined(Ti.App.arguments.launchOptionsLocationKey) || Ti.App.arguments.launchOptionsLocationKey === false))) {
+                    knownBeaconService.refreshBeacons([self.networkId]);
+                }
+            };
+
             // Request authorization is required on IOS devices
             if (OS_IOS) {
-                Ti.Geolocation.requestAuthorization(Ti.Geolocation.AUTHORIZATION_ALWAYS);
-            }
-
-            beaconHandler.init();
-            beaconLog.init();
-            self.setBeaconRegions([]);
-            if (OS_ANDROID || (OS_IOS && (_.isUndefined(Ti.App.arguments.launchOptionsLocationKey) || Ti.App.arguments.launchOptionsLocationKey === false))) {
-                knownBeaconService.refreshBeacons([self.networkId]);
+                var version = Ti.version.split(".");
+                var permissionType = Ti.Geolocation.AUTHORIZATION_ALWAYS;
+                if (version[0] < 5 || // Version < 5
```

Comment:
And then call it version_compare, and put it in it's own library. Or use an [existing version comparing library][1], I guess. ;)

[1]: https://www.npmjs.com/package/compare-version